openshift_tools/monitoring/dockerutil.py

```diff
@@ -196,3 +197,57 @@ def get_ctr_stats(self, ctr, use_cgroups=False):
             raw_stats = self._docker.stats(ctr['Id'], stream=False)
 
         return CgroupUtil.raw_stats_to_dtos(raw_stats)
+
+class CleanupDockerStorage(object):
+    """clean up the docker storage data and metadata"""
+    def __init__(self, docker_client=None):
+        'initial of the client'
+        self.client = docker_client
+        self.all_exited_con_info = None
+        self.all_dangling_img_info = None
+
+        self.exited_con = []
+        self.useless_img = []
+
+    def get_exited_cons(self):
+        'get the exited containers list'
+        self.all_exited_con_info = self.client.containers(all='true', filters={'status': 'exited'})
+        for container in self.all_exited_con_info:
+            self.exited_con.append(container['Id'])
+        return self.exited_con
+
+    def get_nouse_images(self):
+        'get dangling images'
+        self.all_dangling_img_info = self.client.images(filters={'dangling': 'true'})
+        for dangling_img in self.all_dangling_img_info:
+            self.useless_img.append(dangling_img['Id'])
+        return self.useless_img
+
+    def remove_con(self, exited_cons_list):
+        'remove the exited containers from the exited_cons_list if not empty'
+        if len(exited_cons_list) > 0:
+            for con_id in exited_cons_list:
+                try:
+                    self.client.remove_container(container=con_id)
+                except APIError:
+                    pass
```

Comment:
Maybe rather than just 'pass', record that fact that an error occurred while trying to remove the exited containers, and return success/failed from this remove_con() method.